File: benchmarks/numerical/matrix_100.nim
import std/times
const N = 100
var a, b, c: array[N, array[N, float]]
for i in 0..<N:
  for j in 0..<N:
    a[i][j] = float((i*7 + j*13) mod 97)
    b[i][j] = float((i*11 + j*17) mod 89)
let start = cpuTime()
for i in 0..<N:
  for j in 0..<N:
    var sum = 0.0
    for k in 0..<N:
      sum += a[i][k] * b[k][j]
    c[i][j] = sum
echo int((cpuTime() - start) * 1_000_000)

File: benchmarks/numerical/matrix_100.rb
N = 100
a = Array.new(N) { |i| Array.new(N) { |j| ((i*7+j*13) % 97).to_f } }
b = Array.new(N) { |i| Array.new(N) { |j| ((i*11+j*17) % 89).to_f } }
start = Process.clock_gettime(Process::CLOCK_MONOTONIC, :microsecond)
c = Array.new(N) { |i|
  Array.new(N) { |j|
    sum = 0.0
    N.times { |k| sum += a[i][k] * b[k][j] }
    sum
  }
}
finish = Process.clock_gettime(Process::CLOCK_MONOTONIC, :microsecond)
puts(finish - start)
